core/src/main/java/org/bitcoinj/core/NetworkParameters.java

```diff
@@ -267,11 +272,18 @@ public String[] getDnsSeeds() {
     }
 
     /**
-     * Return IP addresses of active peers
-     * @return array of IP addresses
+     * Return socket addresses of active peers
+     * @return array of socket addresses
      */
-    public int[] getAddrSeeds() {
-        return addrSeeds;
+    public final List<InetSocketAddress> getAddrSeeds() {
+        ByteBuffer buf = ByteBuffer.allocate(addrSeeds.length);
+        for (int value : addrSeeds)
+            buf.put((byte) value);
+        ((Buffer) buf).rewind();
+        List<InetSocketAddress> seeds = new LinkedList<InetSocketAddress>();
+        while (buf.hasRemaining())
+            seeds.add(PeerAddress.readSeeds(buf).getSocketAddress());
+        return seeds;
```

Comment:
I would create a `private` helper method (`toInetAddr`) to convert a single `int` to an `InetSocketAddress` and then `getAddrSeeds` can can look  like this:
```
return Stream.of(addrSeeds)
    .map(this:: toInetAddress)
    .collect(StreamUtils.toUnmodifiableList());
```

Reply:
Well, I got my types mixed up above, but you get the idea.